### ISSUE TITLE
bootloop after corrupt suspend without ability to rollback to prev. gen

### ISSUE THREAD

**Issue by denten** [NixOS/nixpkgs#42750]
## Issue description

In changing my configuration.nix I asked systemctl to suspend laptop on low battery (0-5%). Instead, the system began to suspend immediately. Subsequent reboots would wake up the suspended image and immediately suspend again. An attempt to roll back the generations on boot would still bring up the corrupt suspend image and suspend without opportunity for keyboard input. Finally I managed to book from livecd and clear suspend image. This severely breaks the promise of clean roll backs!

I expect selecting any generation but the current should clear suspend image and return to last stable state.

### Steps to reproduce
1. add the following (broken) code to configuration.nix
```
  services.udev.extraRules = ''
       SUBSYSTEM=="power_supply"
       ATTR{status}=="Discharging"
       ATTR{capacity}=="[0-5]"
       RUN+="/run/current-system/sw/bin/systemctl hibernate"
  '';
```

2. Unplug laptop to trigger the discharging state.

## Technical details
  /nix/store/5hszvi0hczzh25mmsz5rkb9qa758c10p-bash-4.4-p12-dev
  /nix/store/bkx6rzqciq0sdn1z5liard33ffvv1n3s-bash-4.4-p12-doc
copying path '/nix/store/bkx6rzqciq0sdn1z5liard33ffvv1n3s-bash-4.4-p12-doc' from 'https://cache.nixos.org'...
copying path '/nix/store/5hszvi0hczzh25mmsz5rkb9qa758c10p-bash-4.4-p12-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.51, NixOS, 18.03.132768.94d80eb7247 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

**Comment by vcunat** [comment]
Correct, there's still no detection whether the chosen generation is the same as the one that created the image.

**Comment by asymmetric** [comment]
@denten could you paste here the steps you used to clear the image, in case other people are bitten by this?

**Comment by dataO1** [comment]
> @denten could you paste here the steps you used to clear the image, in case other people are bitten by this?

Im stuck in the same situation, how do i clear the suspend image?
